src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.cs

```diff
@@ -550,23 +557,40 @@ private static string EntryFromPath(string entry, int offset, int length)
             }
 
             if (length == 0)
-                return String.Empty;
+                return appendPathSeparator ? PathSeparator.ToString() : String.Empty;
```

Comment:
I haven't been able to even exercise the `length == 0` case. I don't think it's possible. *Maybe* on Unix when `sourceDirectoryName` is `"/"`, `includeBaseDirectory` is `true`, and `"/"` doesn't contain any sub directories or files, but a root dir without any sub directories or files seems extremely unlikely (if not impossible), and zipping such a directory seems just as improbable.